### PR TITLE
Enable Kotlin, and upgrade gradle and build tools

### PR DESCRIPTION
**Pull request by bmarty**
Must be merge in parallel with SDK PR https://github.com/matrix-org/matrix-android-sdk/pull/259

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,12 +1,17 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
+    ext.kotlin_version = '1.2.30'
```
[the remark this answers is not in the document]
Comment (bmarty) [reply build.gradle]: I've downgrade to 1.2.30 to match AS plugin version. It is written in f06b1e5 comment :)

File: vector/build.gradle [hunk 2]
```diff
@@ -120,93 +132,95 @@ android {
...
+    kapt 'com.github.bumptech.glide:compiler:4.7.1'
 
-    compile 'com.googlecode.libphonenumber:libphonenumber:8.0.1'
+    implementation 'com.googlecode.libphonenumber:libphonenumber:8.0.1'
 
     /************* Matrix SDK management **************/
     // update settings.gradle
     // use the matrix SDK as external lib
-    compile(name: 'matrix-sdk', ext: 'aar')
+    //implementation(name: 'matrix-sdk', ext: 'aar')
```
Comment (manuroe) [inline vector/build.gradle]: Should be not commented

File: build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcgruenhage) [reply build.gradle]: Ah okay, must have missed that :D

File: build.gradle [hunk 3]
```diff
@@ -27,5 +32,6 @@ allprojects {
         maven {
             url "https://maven.google.com"
```
Comment (Bubu) [inline build.gradle]: This can be removed now as well, right?

Comment (bmarty) [reply build.gradle]: Yes, I'll do this in a future PR

**Pull request merged**